src/java.base/share/man/java.md

```diff
@@ -1292,10 +1292,13 @@ These `java` options control the runtime behavior of the Java HotSpot VM.
 
 `-XX:OnOutOfMemoryError=`*string*
 :   Sets a custom command or a series of semicolon-separated commands to run
-    when an `OutOfMemoryError` exception is first thrown. If the string
+    when an `OutOfMemoryError` exception is first thrown by the JVM.
+    If the string
     contains spaces, then it must be enclosed in quotation marks. For an
     example of a command string, see the description of the `-XX:OnError`
     option.
+    This does not apply to any `OutOfMemoryError` thrown by Java, or native,
+    code in the JDK libraries.
```

Comment:
> This does not apply to any `OutOfMemoryError` thrown by Java, or native, code in the JDK libraries.

This is vague. Does not apply to OOMEs raised explicitly via throw? Or only if thrown from Java code in JDK libraries? So, user code throws are fine? Or, does not apply for OOMEs thrown by JDK libraries regardless of Java or Native? But then, how would a customer know where an exception is thrown without knowing the sources for JDK libraries and hotspot JVM? Why would the throw point even matter?

In my experience, whether or not the OnOOM handling kicks in is a mystery to many customers. We should clearly list the typical points where OnOOM does not work but one would expect it to work, e.g. thread creation errors and explicit throws, regardless of where the throw happens.

Reply:
Okay I have revised in the manner suggested. Thanks

Reply:
Hello David, in this latest updated version, this last sentence:

>This does not apply to any `OutOfMemoryError` thrown by Java, or native, code in the JDK libraries.

appears to be a repeat of the newly added longer explanation one line before. Was it intentional keep this line?

Reply:
Ugghh - a mistake. Thanks.